test/e2e/operator_test.go

```diff
@@ -24,3 +35,129 @@ func TestOperatorNamespace(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// TestPodDisruptionBudgetAtLimitAlert tests that pdb-atlimit alert exists when there is a pdb at limit
+func TestPodDisruptionBudgetAtLimitAlert(t *testing.T) {
+	// See https://bugzilla.redhat.com/show_bug.cgi?id=1762888
+	kubeConfig, err := test.NewClientConfigForTest()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	policyClient, err := policyclientv1beta1.NewForConfig(kubeConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	labels := map[string]string{"app": "pdbtest"}
+	name := names.SimpleNameGenerator.GenerateName("pdbtest-")
+	cmName := names.SimpleNameGenerator.GenerateName("test-configmap-")
+
+	ns, err := createTestNamespace(kubeClient, name)
+	if err != nil {
+		t.Fatalf("could not create test namespace: %v", err)
+	}
+	defer kubeClient.CoreV1().Namespaces().Delete(ns.Name, &metav1.DeleteOptions{})
+
+	err = podCreate(kubeClient, name, labels)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var pods *corev1.PodList
+	// Poll to see if pod is created
+	wait.Poll(time.Second*5, time.Second*300, func() (bool, error) {
+		pods, err = kubeClient.CoreV1().Pods(name).List(metav1.ListOptions{LabelSelector: "app=pdbtest"})
+		if err != nil {
+			return false, err
+		}
+		if len(pods.Items) > 0 {
+			if pods.Items[0].Status.Phase == corev1.PodRunning {
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+
+	err = pdbCreate(policyClient, name, labels)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Wait for pbd
+	err = wait.Poll(time.Second*5, time.Second*300, func() (bool, error) {
+		if _, err := policyClient.PodDisruptionBudgets(name).List(metav1.ListOptions{LabelSelector: "app=pbtest"}); err != nil {
+			return false, fmt.Errorf("Waiting for poddisruptionbudget")
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pq, err := metrics.NewPromQuery(kubeClient, pdbAlertQuery, cmName, ns.Name)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Now check for alert
+	wait.Poll(time.Second*5, time.Second*300, func() (bool, error) {
```

Comment:
wait.InfiniteTestTimeout or something like that instead of 5min

Reply:
same above

Reply:
I've moved to `wait.PollImmediate(time.Second*1, testTimeout, func() (bool, error) {`
and testTimeout 120s.  Test time was 25s (oops didn't realize how long that was) down to 14s.